### PR TITLE
trsting-routes

### PR DESCRIPTION
**Pull request by iiimba**
add route tests + constants
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/18%23discussion_r58245376%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/18%23discussion_r58245630%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/18%23discussion_r58245852%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/18%23discussion_r58245979%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/18%23discussion_r58246150%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/18%23discussion_r58246326%22%5D%2C%20%22comments%22%3A%20%7B%22Pull%204861b3ad4b7f363ead778f50865595eac0abce3c%20DevelopGitHub/GitHubExtension/GitHubExtension.Security.WebApi.Library/Controllers/RepositoryController.cs%209%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/18%23discussion_r58246326%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22ISecyrityCoOntext%20shold%20me%20removed%20from%20injecting%20in%20controller.%20Please%20cooperate%20with%20your%20colleauges%20regarding%20that.%22%2C%20%22created_at%22%3A%20%222016-04-01T18%3A21%3A45Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.Security.WebApi.Library/Controllers/RepositoryController.cs%3AL23-32%22%7D%2C%20%22Pull%204861b3ad4b7f363ead778f50865595eac0abce3c%20DevelopGitHub/GitHubExtension/GitHubExtension.Constant/Role.cs%209%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/18%23discussion_r58245376%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22What%20about%20RoleConstants%3F%20Is%20it%20bettern%20name%20for%20that%20class%3F%22%2C%20%22created_at%22%3A%20%222016-04-01T18%3A15%3A19Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.Constant/Role.cs%3AL1-16%22%7D%2C%20%22Pull%204861b3ad4b7f363ead778f50865595eac0abce3c%20DevelopGitHub/GitHubExtension/GitHubExtension.Security.Tests/TestRoutes/AccountTestRoutes.cs%2018%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/18%23discussion_r58245630%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Do%20you%20have%20some%20feeling%20that%20somthing%20wrong%20with%20this%20code%3F%22%2C%20%22created_at%22%3A%20%222016-04-01T18%3A16%3A56Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.Security.Tests/TestRoutes/AccountTestRoutes.cs%3AL1-73%22%7D%2C%20%22Pull%204861b3ad4b7f363ead778f50865595eac0abce3c%20DevelopGitHub/GitHubExtension/GitHubExtension.Security.WebApi.Library/Controllers/RolesController.cs%2020%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/18%23discussion_r58245979%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Use%20%5BHttpGet%5D%20instead.%20Code%20should%20be%20sel%20documented.%22%2C%20%22created_at%22%3A%20%222016-04-01T18%3A19%3A13Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.Security.WebApi.Library/Controllers/RolesController.cs%3AL18-25%22%7D%2C%20%22Pull%204861b3ad4b7f363ead778f50865595eac0abce3c%20DevelopGitHub/GitHubExtension/GitHubExtension.Security.WebApi.Library/Controllers/RepositoryController.cs%2045%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/18%23discussion_r58246150%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Please%20move%20all%20staff%20related%20to%20ClaimsIdentity%20to%20extension%20methods.%20I%20assumed%20that%20such%20type%20od%20code%20will%20be%20used%20in%20another%20parts%20of%20the%20system.%22%2C%20%22created_at%22%3A%20%222016-04-01T18%3A20%3A30Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.Security.WebApi.Library/Controllers/RepositoryController.cs%3AL51-67%22%7D%2C%20%22Pull%204861b3ad4b7f363ead778f50865595eac0abce3c%20DevelopGitHub/GitHubExtension/GitHubExtension.Constant/RouteConstant.cs%2012%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/18%23discussion_r58245852%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Please%20use%20constants%20first%20letter%20with%20upper%20case.%22%2C%20%22created_at%22%3A%20%222016-04-01T18%3A18%3A24Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.Constant/RouteConstant.cs%3AL1-44%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull 4861b3ad4b7f363ead778f50865595eac0abce3c DevelopGitHub/GitHubExtension/GitHubExtension.Constant/Role.cs 9'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/18#discussion_r58245376'>File: DevelopGitHub/GitHubExtension/GitHubExtension.Constant/Role.cs:L1-16</a></b>
- <a href='https://github.com/TerraVenil'></a> What about RoleConstants? Is it bettern name for that class?
- [ ] <a href='#crh-comment-Pull 4861b3ad4b7f363ead778f50865595eac0abce3c DevelopGitHub/GitHubExtension/GitHubExtension.Security.Tests/TestRoutes/AccountTestRoutes.cs 18'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/18#discussion_r58245630'>File: DevelopGitHub/GitHubExtension/GitHubExtension.Security.Tests/TestRoutes/AccountTestRoutes.cs:L1-73</a></b>
- <a href='https://github.com/TerraVenil'></a> Do you have some feeling that somthing wrong with this code?
- [ ] <a href='#crh-comment-Pull 4861b3ad4b7f363ead778f50865595eac0abce3c DevelopGitHub/GitHubExtension/GitHubExtension.Constant/RouteConstant.cs 12'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/18#discussion_r58245852'>File: DevelopGitHub/GitHubExtension/GitHubExtension.Constant/RouteConstant.cs:L1-44</a></b>
- <a href='https://github.com/TerraVenil'></a> Please use constants first letter with upper case.
- [ ] <a href='#crh-comment-Pull 4861b3ad4b7f363ead778f50865595eac0abce3c DevelopGitHub/GitHubExtension/GitHubExtension.Security.WebApi.Library/Controllers/RolesController.cs 20'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/18#discussion_r58245979'>File: DevelopGitHub/GitHubExtension/GitHubExtension.Security.WebApi.Library/Controllers/RolesController.cs:L18-25</a></b>
- <a href='https://github.com/TerraVenil'></a> Use [HttpGet] instead. Code should be sel documented.
- [ ] <a href='#crh-comment-Pull 4861b3ad4b7f363ead778f50865595eac0abce3c DevelopGitHub/GitHubExtension/GitHubExtension.Security.WebApi.Library/Controllers/RepositoryController.cs 45'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/18#discussion_r58246150'>File: DevelopGitHub/GitHubExtension/GitHubExtension.Security.WebApi.Library/Controllers/RepositoryController.cs:L51-67</a></b>
- <a href='https://github.com/TerraVenil'></a> Please move all staff related to ClaimsIdentity to extension methods. I assumed that such type od code will be used in another parts of the system.
- [ ] <a href='#crh-comment-Pull 4861b3ad4b7f363ead778f50865595eac0abce3c DevelopGitHub/GitHubExtension/GitHubExtension.Security.WebApi.Library/Controllers/RepositoryController.cs 9'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/18#discussion_r58246326'>File: DevelopGitHub/GitHubExtension/GitHubExtension.Security.WebApi.Library/Controllers/RepositoryController.cs:L23-32</a></b>
- <a href='https://github.com/TerraVenil'></a> ISecyrityCoOntext shold me removed from injecting in controller. Please cooperate with your colleauges regarding that.

<a href='https://www.codereviewhub.com/ITsvetkoFF/Kv-013/pull/18?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/ITsvetkoFF/Kv-013/pull/18?approve=1'></a>&nbsp;<a href='https://github.com/ITsvetkoFF/Kv-013/pull/18'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by ITsvetkoFF** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/ITsvetkoFF/Kv-013/pull/18'>Refresh</a>.<a href='#crh-update-none'></a>

File: DevelopGitHub/GitHubExtension/GitHubExtension.Constant/Role.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GitHubExtension.Constant
+{
+    public static class Role
```
Comment (TerraVenil) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Constant/Role.cs]: What about RoleConstants? Is it bettern name for that class?

File: DevelopGitHub/GitHubExtension/GitHubExtension.Security.Tests/TestRoutes/AccountTestRoutes.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+using System.Text.RegularExpressions;
+
+namespace GitHubExtension.Security.Tests.TestRoutes
+{
+    public class AccountTestRoutes : TestRoutesConfig
+    {
+        [Fact]
+        public void AccountGetUserTest()
+        {
+            url = "/" + RouteConstant.apiAccount +
```
Comment (TerraVenil) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Security.Tests/TestRoutes/AccountTestRoutes.cs]: Do you have some feeling that somthing wrong with this code?

File: DevelopGitHub/GitHubExtension/GitHubExtension.Constant/RouteConstant.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GitHubExtension.Constant
+{
+    public class RouteConstant
+    {
+        // route prefixes
+        public const string apiRoles = "api/v1/roles";
```
Comment (TerraVenil) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Constant/RouteConstant.cs]: Please use constants first letter with upper case.

File: DevelopGitHub/GitHubExtension/GitHubExtension.Security.WebApi.Library/Controllers/RolesController.cs [hunk 4]
```diff
@@ -17,7 +18,7 @@ public RolesController(ISecurityContext securityContext)
 
         private readonly ISecurityContext _securityContext;
         
-
+        // GET
```
Comment (TerraVenil) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Security.WebApi.Library/Controllers/RolesController.cs]: Use [HttpGet] instead. Code should be sel documented.

File: DevelopGitHub/GitHubExtension/GitHubExtension.Security.WebApi.Library/Controllers/RepositoryController.cs [hunk 5]
```diff
@@ -49,15 +51,16 @@ public RepositoryController(IGithubService guGithubService, ISecurityContext sec
...
 
+        // GET
         [Authorize]
-        [Route("api/repos/{repoName}/collaborators")]
-        public async Task<IHttpActionResult> GetCollaboratorsForRepo(string repoName)
+        [Route(RouteConstant.getCollaboratorsForRepository)]
+        public async Task<IHttpActionResult> GetCollaboratorsForRepo(string repositoryName)
         {
             var claims = User.Identity as ClaimsIdentity;
             string token = claims.FindFirstValue("ExternalAccessToken");
```
Comment (TerraVenil) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Security.WebApi.Library/Controllers/RepositoryController.cs]: Please move all staff related to ClaimsIdentity to extension methods. I assumed that such type od code will be used in another parts of the system.

File: DevelopGitHub/GitHubExtension/GitHubExtension.Security.WebApi.Library/Controllers/RepositoryController.cs [hunk 6]
```diff
@@ -22,9 +23,9 @@ public RepositoryController(IGithubService guGithubService, ISecurityContext sec
             _securityContext = securityContext;
```
Comment (TerraVenil) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Security.WebApi.Library/Controllers/RepositoryController.cs:23]: ISecyrityCoOntext shold me removed from injecting in controller. Please cooperate with your colleauges regarding that.

File: DevelopGitHub/GitHubExtension/GitHubExtension.Security.Tests/TestRoutes/AccountTestRoutes.cs (hunk 2 above)
Comment (TerraVenil) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Security.Tests/TestRoutes/AccountTestRoutes.cs]: So I suggest than move symbol "/" or to base class or extension method for asserting routes...

**Pull request closed without merging**